File: abms-backend/auth-service/src/main/java/com/abms/auth/service/AuthService.java
package com.abms.auth.service;

import com.abms.auth.dto.AuthResponse;
import com.abms.auth.dto.ChangePasswordRequest;
import com.abms.auth.dto.LoginRequest;
import com.abms.auth.dto.RegisterRequest;
import java.util.List;
import java.util.UUID;

public interface AuthService {

    AuthResponse register(RegisterRequest request);

    AuthResponse login(LoginRequest request);

    AuthResponse changePassword(String authorizationHeader, ChangePasswordRequest request);

    List<AuthResponse> getPendingResidents();

    AuthResponse approveResident(UUID userId);

    AuthResponse rejectResident(UUID userId);
}

File: abms-frontend/frontend/src/services/authService.js
import { apiClient } from './apiClient.js'

export const authService = {
  async login(payload) {
    const { data } = await apiClient.post('/api/v1/auth/login', payload)
    return data
  },

  async changePassword(payload) {
    const { data } = await apiClient.post('/api/v1/auth/change-password', payload)
    return data
  },

  async register(payload) {
    const { data } = await apiClient.post('/api/v1/auth/register', payload)
    return data
  },

  async getPendingResidents() {
    const { data } = await apiClient.get('/api/v1/auth/residents/pending')
    return data
  },

  async approveResident(userId) {
    const { data } = await apiClient.post(`/api/v1/auth/residents/${userId}/approve`)
    return data
  },

  async rejectResident(userId) {
    const { data } = await apiClient.post(`/api/v1/auth/residents/${userId}/reject`)
    return data
  },
}